pkg/eks/components/AccountAccess.vue

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { _CREATE, _VIEW } from '@shell/config/query-params';
+import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import SelectCredential from '@shell/edit/provisioning.cattle.io.cluster/SelectCredential.vue';
+import { DEFAULT_REGION } from './CruEKS.vue';
+import { mapGetters } from 'vuex';
+
+export default defineComponent({
+  name: 'EKSAccountAccess',
+
+  components: {
+    LabeledSelect,
+    SelectCredential
+  },
+
+  props: {
+    mode: {
+      type:    String,
+      default: _CREATE
+    },
+
+    credential: {
+      type:    String,
+      default: null
+    },
+
+    region: {
+      type:    String,
+      default: ''
+    }
+  },
+
+  async fetch() {
+    this.defaultRegions = await this.$store.dispatch('aws/defaultRegions');
+    if (this.defaultRegions.length && !this.region) {
+      (this.$emit('update-region', DEFAULT_REGION));
+    }
+  },
+
+  data() {
+    return { regions: [] as any[], defaultRegions: [] as string[] };
+  },
+
+  watch: {
+    isAuthenticated: {
+      async handler(neu) {
+        if (neu) {
+          await this.fetchRegions();
+        } else {
+          if (this.defaultRegions.length && !this.defaultRegions.includes(this.region)) {
+            if (this.defaultRegions.includes(DEFAULT_REGION)) {
+              this.$emit('update-region', DEFAULT_REGION);
+            } else {
+              this.$emit('update-region', this.defaultRegions[0]);
+            }
+          }
+        }
+      },
+      immediate: true
+    }
+  },
+
+  methods: {
+    async fetchRegions() {
+      const { region, credential } = this as any;
```

Comment:
to avoid as any, this can be `const { region, credential}: { region: string, credential: string} = this`